src/main/java/com/atlassian/bitbucket/jenkins/internal/trigger/BitbucketWebhookConsumer.java

```diff
@@ -98,7 +73,11 @@ private void triggerJob(RefsChangedWebhookEvent event,
 
     private static Collection<? extends SCM> getScms(ParameterizedJobMixIn.ParameterizedJob<?, ?> job) {
         SCMTriggerItem triggerItem = SCMTriggerItem.SCMTriggerItems.asSCMTriggerItem(job);
-        if (triggerItem != null) {
+        if (triggerItem instanceof WorkflowJob) {
+            if (((WorkflowJob) triggerItem).getDefinition() instanceof CpsScmFlowDefinition) {
+                return Collections.singletonList(((CpsScmFlowDefinition) ((WorkflowJob) triggerItem).getDefinition()).getScm());
+            }
+        } else if (triggerItem != null) {
```

Comment:
Quickly looking into `BitbucketScmRunHelper`, there is a method `getBitbucketScm`. I think we should have a central way of fetching scm from workflow job.

Reply:
I agree, it's a good idea to consolidate this getter, given we'll need to repeat this pattern. I'll add the logic to a separate method in the `BitbucketScmRunHelper` with javadoc :)

Reply:
Did not notice this! We can (and should) use this instead.

Reply:
Are there any instances where `getScms` will return multiple Scms and you need each? If not we can actually omit this method entirely in favour of that one.

Reply:
Would like to talk about this more. Right now it has private access, but I agree we use this often enough that a centralized helper method should do it.

Reply:
I'm keen to release today so will have this discussion later, but merge this.